docs/book/features/vsphere_csi_migration.md

```diff
@@ -0,0 +1,122 @@
+3. upgrade Kubernetes cluster to 1.19 and enable the feature flags `CSIMigration` and `CSIMigrationvSphere` on `kube-controller` and `kubelet` on all controller and workload nodes.
+
+   - `CSIMigrationvSphere` flag enables shims and translation logic to route volume operations from the vSphere in-tree plugin to vSphere CSI plugin. Supports falling back to in-tree vSphere plugin if a node does not have vSphere CSI plugin installed and configured. This flag requires `CSIMigration` feature flag to be enabled.
+
+   - update kube-controller-manager manifest file.
+This file is generally available at `/etc/kubernetes/manifests/kube-controller-manager.yaml`
+
+        - --feature-gates=CSIMigration=true,CSIMigrationvSphere=true
+
+   - update kubelet configuration file. This file is generally available at `/var/lib/kubelet/config.yaml`
+
+          featureGates:
+          {
+            "CSIMigration": true,
+            "CSIMigrationvSphere": true
+          }
+
+Please note that during a cluster upgrade we must drain each node (remove running workloads) before changing the configuration on the Kubelet.
+
+There is also an optional `CSIMigrationvSphereComplete` flag that can be enabled if all of the nodes have CSI migration enabled.
+`CSIMigrationvSphereComplete` helps stop registering the vSphere in-tree plugin in kubelet and volume controllers and enables shims and translation logic to route volume operations from the vSphere in-tree plugin to vSphere CSI plugin. Requires `CSIMigration` and `CSIMigrationvSphere` feature flags enabled and vSphere CSI plugin installed and configured on all nodes in the cluster.
+
+To verify the `CSIMigrationvSphere` feature flag is enabled and vSphere CSI driver installed on a particular node you can get the CSINode object. You should see `kubernetes.io/vsphere-volume` for an annotation `storage.alpha.kubernetes.io/migrated-plugins`.
+
+    # kubectl get csinodes -o yaml
+    apiVersion: v1
+    items:
+    - apiVersion: storage.k8s.io/v1
+      kind: CSINode
+      metadata:
+        annotations:
+          storage.alpha.kubernetes.io/migrated-plugins: kubernetes.io/vsphere-volume
+
+To verify vSphere in-tree PVCs and PVs are migrated to vSphere CSI driver, verify `pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com` annotations is present on PVCs and PVs.
+
+PVC Annotations
+
+        Annotations:   pv.kubernetes.io/bind-completed: yes
+                       pv.kubernetes.io/bound-by-controller: yes
+                       pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com
+                       volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/vsphere-volume
+
+PV Annotations
+
+        Annotations:     kubernetes.io/createdby: vsphere-volume-dynamic-provisioner
+                         pv.kubernetes.io/bound-by-controller: yes
+                         pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com
+                         pv.kubernetes.io/provisioned-by: kubernetes.io/vsphere-volume
+
```

Comment:
In addition to annotation, can you also show how migrated PV looks different from PV provisioned by CSI driver?

Reply:
the only difference I see is the following. Will add 

```
New in-tree vSphere volumes created by vSphere CSI driver can be identified by following annotations. PV spec will still hold vSphere Volume Path, so in case if migration needs to be disabled. Provisioned volume can be used by in-tree vSpher plugin.


PVC Annotations

    Annotations:   pv.kubernetes.io/bind-completed: yes
                   pv.kubernetes.io/bound-by-controller: yes
                   volume.beta.kubernetes.io/storage-provisioner: csi.vsphere.vmware.com

PV Annotations

    Annotations:     pv.kubernetes.io/provisioned-by: csi.vsphere.vmware.com
```